### PR TITLE
Elasticsearch backports from upstream

### PR DESCRIPTION
**Pull request by kohaputti**
This is a pull request for backporting elasticsearch patches from community.

### CODE REVIEW

**Comment by kohaputti** [comment]
Intranet seems to be working now fine, SRU is still to be tested.

**Comment by kohaputti** [comment]
Ok, SRU should be fine, too. Waiting for someone now to review the changes

File: installer/data/mysql/atomicupdate/bug_19575-use-canonical-field-names-and-resolve-aliased-fields.sql [hunk 1]
```diff
@@ -7,7 +7,7 @@ UPDATE `search_field` SET `name` = 'control-number', `label` = 'control-number'
 UPDATE `search_field` SET `name` = 'place-of-publication', `label` = 'place-of-publication' WHERE `name` = 'place';
 UPDATE `search_field` SET `name` = 'date-of-acquisition', `label` = 'date-of-acquisition' WHERE `name` = 'acqdate';
 UPDATE `search_field` SET `name` = 'isbn', `label` = 'isbn' WHERE `name` = 'isbn';
-UPDATE `search_field` SET `name` = 'authority-number', `label` = 'authority-number' WHERE `name` = 'an';
+UPDATE `search_field` SET `name` = 'koha-auth-number', `label` = 'koha-auth-number' WHERE `name` = 'an';
```
Comment (johannaraisa) [inline installer/data/mysql/atomicupdate/bug_19575-use-canonical-field-names-and-resolve-aliased-fields.sql]: Is our atomicupdate able run these sql scripts? This is different from our the files we use.

File: installer/data/mysql/atomicupdate/bug_20334-add-syspref-QueryRegexEscapeOptions.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+INSERT IGNORE INTO systempreferences ( `variable`, `value`, `options`, `explanation`, `type`) VALUES  ('QueryRegexEscapeOptions', 'dont_escape', 'dont_escape|escape|unescape_escaped', 'Escape option for regexps delimiters in Elasicsearch queries.', 'Choice');
```
Comment (johannaraisa) [inline installer/data/mysql/atomicupdate/bug_20334-add-syspref-QueryRegexEscapeOptions.sql]: This also doesn't match to atomicupdate files that we use.

File: installer/data/mysql/atomicupdate/bug_19575-use-canonical-field-names-and-resolve-aliased-fields.sql (hunk 1 above)
Comment (kohaputti) [reply installer/data/mysql/atomicupdate/bug_19575-use-canonical-field-names-and-resolve-aliased-fields.sql]: I got it to work and the stuff is in my dev install database but now I notice that this should also go to kohastructure.sql. Will submit a patch for that.

**Comment by kohaputti** [comment]
Ok, now the QueryRegexEscapeOptions syspref is also added on a new installation. Ready to be reviewed and then merged!

**Comment by EreMaijala** [comment]
Looks good to me.

**Pull request merged**